azurerm/data_source_cosmos_db_account.go

```diff
@@ -0,0 +1,218 @@
+			"offer_type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"kind": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"ip_range_filter": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"enable_automatic_failover": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+
+			"consistency_policy": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"consistency_level": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"max_interval_in_seconds": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+
+						"max_staleness_prefix": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"geo_location": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+
+						"id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"location": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"failover_priority": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"endpoint": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"read_endpoints": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"write_endpoints": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"primary_master_key": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"secondary_master_key": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"primary_readonly_master_key": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"secondary_readonly_master_key": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceArmCosmosDBAccountRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).cosmosDBClient
+	ctx := meta.(*ArmClient).StopContext
+
+	resourceGroup := d.Get("resource_group_name").(string)
+	name := d.Get("name").(string)
+
+	resp, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error making Read request on AzureRM CosmosDB Account '%s': %s", name, err)
+	}
+
+	d.SetId(*resp.ID)
+	d.Set("name", resp.Name)
+	d.Set("location", azureRMNormalizeLocation(*resp.Location))
+	d.Set("resource_group_name", resourceGroup)
+	flattenAndSetTags(d, resp.Tags)
+
+	d.Set("kind", string(resp.Kind))
+	d.Set("offer_type", string(resp.DatabaseAccountOfferType))
+	d.Set("ip_range_filter", resp.IPRangeFilter)
+	d.Set("endpoint", resp.DocumentEndpoint)
```

Comment:
so we should be accessing all of these properties via the `Properties` sub-object, since this matches the actual API response (e.g. `resp.Properties.DocumentEndpoint`); and also check for this being nil via:

```
if props := resp.[XX]Properties; props != nil {
  d.Set("ip_range_filter", props.IPRangeFilter)
}
```

I've been meaning to open a bug on the Azure SDK repository about removing the top-level fields, since they're duplicating the other values and not consistent with the API Response (with the exception of Name, Location, Kind which need to be at the top-level)